java-checks/src/main/java/org/sonar/java/checks/SynchronizedClassUsageCheck.java

```diff
@@ -130,18 +131,34 @@ public void visitClass(ClassTree tree) {
     @Override
     public void visitMethod(MethodTree tree) {
       TypeTree returnTypeTree = tree.returnType();
-      if (isNotOverriding(tree) || returnTypeTree == null) {
+      populateOverridingMethodExclusions(tree, returnTypeTree);
+      if (!isOverriding(tree) || returnTypeTree == null) {
```

Comment:
Why is data implicitly pushed to the stack but explicitly popped?
```suggestion
      overridingMethodTypes.push(collectTypesFromOverridingMethodSignature(tree, returnTypeTree));
      if (!isOverriding(tree) || returnTypeTree == null) {
```